haystack/components/routers/metadata_router.py

```diff
@@ -120,7 +120,7 @@ def run(self, documents: Union[list[Document], list[ByteStream]]):
         """
 
         unmatched = []
-        output: dict[str, Union[list[Document], list[ByteStream]]] = {edge: [] for edge in self.rules}
+        output: dict[str, Union[list[Document], list[ByteStream]]] = {edge: [] for edge in self.rules}  # type: ignore
```

Comment:
if we do go with the type ingore which would be easiest for now, le'ts also add the reason like `# type: ignore[xxx]` where `xxx` is the reason mypy gives